### PR TITLE
refactor: Update webpack to 4.6.0

### PR DESCRIPTION
**Pull request by ffflorian**
Closes #1410.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,41 +1,34 @@
...
-    })(),
-    entry: path.resolve(__dirname, 'electron/renderer/src/index.js'),
-    module: {
-      rules: [
-        {
-          exclude: /(node_modules)/,
-          test: /\.js$/,
-          use: ['babel-loader'],
+module.exports = (env = {}) => ({
+  devtool: env.production ? '' : 'cheap-eval-source-map',
```
Comment (Yserz) [inline webpack.config.js]: I would prefer `undefined` instead of `''`

File: webpack.config.js [hunk 2]
```diff
@@ -1,41 +1,34 @@
...
+  entry: path.resolve(__dirname, 'electron/renderer/src/index.js'),
+  mode: env.production ? 'production' : 'development',
+  module: {
+    rules: [
+      {
+        exclude: /(node_modules)/,
```
Comment (Yserz) [inline webpack.config.js]: `(` and `)` can be removed

File: webpack.config.js [hunk 3]
```diff
@@ -1,41 +1,34 @@
...
-    module: {
-      rules: [
-        {
-          exclude: /(node_modules)/,
-          test: /\.js$/,
-          use: ['babel-loader'],
+module.exports = (env = {}) => ({
+  devtool: env.production ? '' : 'cheap-eval-source-map',
+  entry: path.resolve(__dirname, 'electron/renderer/src/index.js'),
+  mode: env.production ? 'production' : 'development',
```
Comment (Yserz) [inline webpack.config.js]: What do you think about inverting the condition to make production the default in case of an error?

Comment (ffflorian) [reply webpack.config.js]: Good idea

**Comment by ffflorian** [comment]
Can I merge or should we wait for automation?

**Comment by gregor** [comment]
Good to be merged once the travis run is green.

**Comment by Yserz** [comment]
Merge it! It's green now :)

**Comment by lipis** [comment]
Next one is Electron 2.0 :D

**Pull request merged**